contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -185,10 +220,17 @@ func (s *SubscriptionsSupervisor) UpdateSubscriptions(channel *eventingv1alpha1.
 		for sub := range chMap {
 			s.unsubscribe(cRef, sub)
 		}
-		delete(s.subscriptions, cRef)
+		s.deleteChannel(cRef)
```

Comment:
this seems like a weird update. seems like the channel will be deleted if you create it but have no existing subscriptions. How will this support late binding?

Reply:
I made this (somewhat contrived) change just to ensure there are no memory leaks by consistently calling a function that would cleanup both the `subscriptions` and `reconcilers` maps.

This may be due to my relative inexperience with Go, but I'm not sure what you mean by late binding in this context. Can you expand on that?